bin/add_frontend_server.rb

```diff
@@ -1,16 +1,61 @@
 require 'aws-sdk'
 require_relative '../deployment'
+require 'net/ssh'
+require 'net/scp'
+require 'io/console'
+require 'json'
+require 'etc'
 
 MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
 
+# Executes an arbitrary command on an ssh channel, prints the output to console, bails out if the command fails
+# channel: ssh channel that you get from open_channel
+# command: Command to execute on the remote host
+# exit_error_string: Error message to throw if the command exits with something other than 1
+def execute_ssh_on_channel(channel, command, exit_error_string)
+  channel.exec(command) do |ch|
+    ch.on_data do |_, data|
+      $stdout.print data
+    end
+
+    ch.on_extended_data do |_, _, data|
+      $stderr.print data
+    end
+
+    ch.on_request 'exit-status' do |_, data|
+      if data.read_long != 0
+        throw exit_error_string
+      end
+    end
+  end
+
+end
+
+options = {}
+username = Etc.getlogin
+
+OptionParser.new do |opts|
+  opts.on('-e', '--environment ENVIRONMENT', 'Environment to add frontend to') do |env|
+    options['environment'] = env
+  end
+
+  opts.on('-h', '--help', 'Print this') { puts options; exit }
+end.parse!
+
+raise OptionParser::MissingArgument, 'Environment is required' if options['environment'].nil?
+
+Net::SSH.start('gateway.code.org', username) do |ssh|
+  puts ssh.exec!('echo \'Verifying connection to gateway\'')
+end
+
 ec2client = Aws::EC2::Client.new
 
 instances = ec2client.describe_instances
 
 #Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
 frontend_instances = instances.reservations.map do |reservation|
-  reservation.instances.select{|instance| instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
-end
+  reservation.instances.select{|instance| instance.state.name == 'running' && instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
+ end
```

Comment:
Lint/BlockAlignment: `end` at 58, 1 is not aligned with `frontend_instances = instances.reservations.map do |reservation|` at 56, 0